modules/lib/core/index.d.ts

```diff
@@ -52,23 +67,115 @@ export interface PublishInfo {
     first?: string;
 }
 
-export interface Component<Config extends object = object, Regions extends Record<string, Region> = Record<string, Region>> {
-    config: Config;
-    descriptor: string;
+export interface FragmentComponent {
+    type: 'fragment'
+    fragment: string;
     path: string;
-    type: 'page' | 'layout' | 'part';
+}
+
+export interface LayoutComponent<
+    Descriptor extends ComponentDescriptor = ComponentDescriptor,
+    Config extends NestedRecord = NestedRecord,
+    Regions extends
+        Record<string, Region<(FragmentComponent | PartComponent | TextComponent)[]>> = 
+        Record<string, Region<(FragmentComponent | Part          | TextComponent)[]>>
+> {
+    type: 'layout'
+    descriptor: Descriptor
+    config: Config
+    path?: string // Missing in fragmentPreview https://github.com/enonic/xp/issues/10116
+    regions: Regions;
+}
+type LayoutDescriptor = keyof XpLayoutMap;
+type Layout = LayoutDescriptor extends any // this lets us iterate over every member of the union
+    ? LayoutComponent<LayoutDescriptor>
```

Comment:
Ok, this wasn't thought through. Since LayoutComponent doesn't look up config from XpLayoutMap, this won't work